src/datoms_differ/datom.cljc

```diff
@@ -116,39 +115,34 @@
                c#)))
          res))))
 
+(defprotocol DatomValueComparator
+  "A protocol that allows you to provide custom compare for types of your choosing for the value attribute of a datom"
+  (customCompareTo [this other]))
+
 (defn cmp [o1 o2]
   (if (nil? o1) 0
       (if (nil? o2) 0
           (compare o1 o2))))
 
+(defn custom-value-cmp [v1 v2]
+  ;; satisfies? would have made sense here, but it's painfully slow in such a hot path as this.
+  ;; There's a ticket waiting to address this: https://clojure.atlassian.net/browse/CLJ-1814
+  (if (extends? DatomValueComparator (type v1))
+    (customCompareTo v1 v2)
+    (cmp v1 v2)))
+
```

Comment:
How about dropping this function altogether, and rather supply a default implementation (i.e., using cmp) of customCompareTo for java.lang.Obect / js/object?

Reply:
Not sure I understand. `cmp` calls clojure `compare` which requires that the object implements `java.util.Comparable` no? 

Ahh... so that would result in an end result as if you haven't done anything, but you avoid the `extends?` check all together? Am I onto something ? :)

Reply:
Yes. Let Clojure do the polymorphic dispatching instead of you doing so via extend?.